.github/workflows/ci-build.yaml

```diff
@@ -54,3 +54,8 @@
       - name: Build
         run: |
           make
+
+      - uses: actions/upload-artifact@v3
```

Comment:
Fix this alert by pinning to a hash instead of a version number.  As noted in the code scanning warning details:

> For GitHub workflows, pin dependencies by hash. See [main.yaml](https://github.com/ossf/scorecard/blob/f55b86d6627cc3717e3a0395e03305e81b9a09be/.github/workflows/main.yml#L27) for example. To determine the permissions needed for your workflows, you may use [StepSecurity's online tool](https://app.stepsecurity.io/) by ticking the "Pin actions to a full length commit SHA". You may also tick the "Restrict permissions for GITHUB_TOKEN" to fix issues found by the Token-Permissions check.